### PR TITLE
regression: Don't encrypt attachment description for disabled E2EE rooms

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Proposed changes (including videos or screenshots)

Implemented proper checks before encrypting message on file upload flow.
## Steps to test or reproduce

1. Create an E2EE room, then disable E2EE in the room.
2. Send some attachment with description
3. The message is encrypted (should not be encrypted)

## Further comments

[E2EE-26](https://rocketchat.atlassian.net/browse/E2EE-26)

[E2EE-26]: https://rocketchat.atlassian.net/browse/E2EE-26?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]
Can we create some test cases to cover this?

File: apps/meteor/client/lib/chats/flows/uploadFiles.ts [hunk 1]
```diff
@@ -61,7 +63,11 @@ export const uploadFiles = async (chat: ChatAPI, files: readonly File[], resetFi
 						return;
 					}
 
-					const shouldConvertSentMessages = e2eRoom.shouldConvertSentMessages({ msg });
+					const subscription = await waitUntilFind(() => ChatRoom.findOne({ _id: room._id }));
```
Comment (gabriellsh) [inline apps/meteor/client/lib/chats/flows/uploadFiles.ts:66]: The `room` object declared on line 19 should be the same as this. Is it possible to use it instead?

**Review by gabriellsh — CHANGES_REQUESTED** [review]

File: apps/meteor/client/lib/chats/flows/uploadFiles.ts [hunk 2]
```diff
@@ -61,7 +61,9 @@ export const uploadFiles = async (chat: ChatAPI, files: readonly File[], resetFi
 						return;
 					}
 
-					const shouldConvertSentMessages = e2eRoom.shouldConvertSentMessages({ msg });
+					room.encrypted ? e2eRoom.resume() : e2eRoom.pause();
```
Comment (sampaiodiego) [inline apps/meteor/client/lib/chats/flows/uploadFiles.ts:64]: did we starting executing code like this? I've always suggested to do the following, it is much cleaner IMO.. ternary should be used only attributions..
```suggestion
					if (room.encrypted) {
						e2eRoom.resume();
					} else {
						e2eRoom.pause();
					}
```

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: apps/meteor/client/lib/chats/flows/uploadFiles.ts (hunk 2 above)
Comment (debdutdeb) [reply apps/meteor/client/lib/chats/flows/uploadFiles.ts:64]: +1 from me. Goes for both ternaries and &&s, ||s (room.encrypted && e2eRoom.resume()). 

Also reduces more noise down the road in diffs.

Comment (yash-rajpal) [reply apps/meteor/client/lib/chats/flows/uploadFiles.ts]: Agreed. Done!

In my defense, I copied this piece of code from someplace else, hence the negligence. 😬 

Will keep this in mind moving forward, thanks!

File: apps/meteor/client/lib/chats/flows/uploadFiles.ts (hunk 1 above)
Comment (yash-rajpal) [reply apps/meteor/client/lib/chats/flows/uploadFiles.ts]: Done!

**Comment by yash-rajpal** [comment]
This PR solved a regression caused by encryption of file description - #31840 , we reverted the encryption of file description feature - #32324 with the idea of delivering full file encryption in next release, so this regression no longer makes sense as the main PR was reverted.

**Pull request closed without merging**